### ISSUE TITLE
flatpak Thunderbird won't save multiple attachments to a directory the user has access to

### ISSUE THREAD

**Issue by hzulla** [flathub/org.mozilla.Thunderbird#266]
This happened using Thunderbird 102.2.2 on Ubuntu 22.04 with flatpak `1.14.0-2flatpak1~22.04.1` (installed via the ppa).

- I have a message in my sent folder that has 13 PDF documents attached to it
- Wanted to save all PDF files into a local folder
- Clicked "Save All" button (bottom right corner)
- TB offered me a file picker dialog to choose the target directory
- I chose the `/tmp/` folder, file picker dialog closes and returns to TB
- TB responds with two modal error message dialogs:
  > "Unable to save the attachment. Please check your file name and try again later." [OK]
  > "Unable to save the message. Please check your file name and try again later." [OK]
- That's it, none of the 13 files were saved to `/tmp/` despite having write access to it.

**Comment by Erick555** [comment]
Weird, it works for me. Does it always happens for all attachments?

**Comment by hzulla** [comment]
Yes.

![tb](https://user-images.githubusercontent.com/1705654/191592145-0c2cf27a-63e7-45d1-9ac6-707a099b67b5.png)

If I click on the "Save All" button instead (marked red in the screenshot above), the file chooser appears, I can choose the same target location directory and the reported behaviour of the bug report happens. This happens with any message that contains multiple attachments that I tried.

**Comment by hzulla** [comment]
It works for messages that contain only one attachment. The button shown in the screenshot then changes from `[Save all]` to `[Save]` and works as expected.

**Comment by hzulla** [comment]
Brief detour, possibly related to this bug:

1. In Thunderbird, if I click on an individual attachment to save it, the file chooser appears, I can choose a target location directory such as `/tmp`, change the target filename if desired, and in the end the file is stored at the chosen target location as expected.
2. If I then click on a different attachment right after that (or use the `[Save]` button multiple times), [the bug reported here](https://bugzilla.mozilla.org/show_bug.cgi?id=1785158) happens - the file chooser doesn't show me the `/tmp` directory with all of its files as the target directory for the second file, but instead shows me a virtual `/run/user/1000/doc/<hex_id_number>` directory that only contains the file I stored in step 1. As a naive user I assumed that the second file would now be stored into the same directory from step 1, but while I can select this target location in the file chooser and TB (or FF) do not complain and show no error message about it, the second file is never stored anywhere and just silently dropped.

**Comment by Erick555** [comment]
So it doesn't work because you try saving file to `/run/user/1000/doc/` instead of actual `/tmp`?

**Comment by j-wiedemann** [comment]
Hi !
Same problem here with TB 102.3.0 (64 bits).
But I notice that the save all function work only if I save files somewhere inside ~/Downloads.

**Comment by hzulla** [comment]
> when you change path in file dialog back to `/tmp` it works??

Yes, that is true for the "detour bug report" that became https://github.com/flatpak/xdg-desktop-portal/issues/886

However, sorry for the confusion, the original bug report in this issue here is about the `[Save All]` button. I can confirm @j-wiedemann's report that "save all" with a multiple-attachments-message works for me if I choose `~/Downloads` as the target folder, but doesn't work if I choose `/tmp` in the file chooser.

**Comment by Erick555** [comment]
So this issue isn't the same as the one you reported in the x-d-p repo or it is?

**Comment by hzulla** [comment]
No, the bug report in the first entry in this issue is a different bug than the one later down below. I rephrased the latter into a separate bug report in the x-d-p repo.

However, I suspect these two bugs are possibly related, since they both occur when one tries to save more than one file at a time to a directory other than `~/Downloads`.
